test/unit/spec/nbextensions/appCell2/widgets/appCellWidget-spec.js

```diff
@@ -119,26 +118,39 @@ define([
         });
 
         it('Has expected functions when instantiated', () => {
-            expect(mockAppCell.init).toBeDefined();
-            expect(mockAppCell.attach).toBeDefined();
-            expect(mockAppCell.start).toBeDefined();
-            expect(mockAppCell.stop).toBeDefined();
-            expect(mockAppCell.detach).toBeDefined();
+            ['init', 'attach', 'start', 'stop', 'detach'].forEach(fn => {
+                expect(mockAppCell[fn]).toBeDefined();
+            });
         });
 
-        it('has a method "init" which returns a promise then null', async () => {
-            var initPromise = mockAppCell.init();
-            expect(initPromise instanceof Promise).toBeTrue();
+        it('has a method "init" which returns a promise then null', () => {
+            return mockAppCell.init()
```

Comment:
Gotta return the promises, otherwise they get dangled and not always consumed right, especially with Jasmine 3.6.0+. This spec failed consistently before this change.

Reply:
Yup -- it was also marked as `async` previously but didn't have a matching `await`. ESLint should have picked that up. 😖